src/Context/cart-context.js

```diff
@@ -0,0 +1,179 @@
+import axios from "axios";
+import { createContext, useContext, useReducer, useEffect } from "react";
+import cartReducer from "../reducer/cartReducer";
+import { useAuth } from "./auth-context";
+
+const CartContext = createContext();
+const initialState = [];
+
+const CartProvider = ({ children }) => {
+  const [cartState, cartDispatch] = useReducer(cartReducer, initialState);
+  const { authState } = useAuth();
+  const { token } = authState;
+
+  const getCart = async () => {
+    try {
+      const config = {
+        headers: {
+          authorization: token,
+        },
+      };
+      const response = await axios.get("/api/user/cart", config);
+      if (response.status == 200) {
+        cartDispatch({
+          type: "INITIALIZE",
+          payload: { products: response.data.cart },
+        });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const alreadyInCart = (_id) => {
+    let flag = false;
+    cartState.map((item) => {
+      if (item._id === _id) {
+        flag = true;
+      }
+    });
+    return flag;
+  };
+
+  const addToCart = async (product) => {
+    try {
```

Comment:
you can extract these functions in context for clean code. otherwise it's coded well.